strings-csharp/ReplaceSpecialCharactersInString/ReplaceSpecialCharactersInString/StringReplacer.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+using BenchmarkDotNet.Attributes;
+
+namespace ReplaceSpecialCharactersInString
+{
+    [MemoryDiagnoser]
+    public class StringReplacer
+    {
+        [Benchmark]
+        public static string ReplaceUsingStringReplace(string originalString , char charToBeReplaced, char charToReplace)
+        {
+            return originalString.Replace(charToBeReplaced.ToString(), charToReplace.ToString());
+        }
+
+
+        [Benchmark]
+        public static string ReplaceUsingStringBuilder(string originalString , char charToBeReplaced, char charToReplace)
+        {
+            var sb = new StringBuilder(originalString);
+
+            sb.Replace(charToBeReplaced.ToString(), charToReplace.ToString());
+
+            return sb.ToString();
+        }
+
+
+        [Benchmark]
+        public static string ReplaceUsingRegex(string originalString, char charToBeReplaced, char charToReplace)
+        {
+            var pattern = $"[{Regex.Escape(charToBeReplaced.ToString())}]";
+
+            return Regex.Replace(originalString, pattern, charToReplace.ToString());
+        }
+
+
+        [Benchmark]
+        public static string ReplaceUsingSpan(string originalString , char charToBeReplaced, char charToReplace)
+        {
+            var originalChars = originalString.ToCharArray();
+            var span = new Span<char>(originalChars);
+
+            for (int i = 0; i < originalString.Length; i++)
+            {
+                if (span[i] == charToBeReplaced)
+                    span[i] = charToReplace;
+            }
+
+            return new string(span.ToArray());
+        }
+
+
+        [Benchmark]
+        public static string ReplaceUsingInefficientMultipleReplacementsStringReplace(string originalString , char charToBeReplaced, char charToReplace)
```

Comment:
The purpose is to demonstrate that it is a bad practice to do so and in case we need to call multiple replace on different characters it is wise to consider other methods like string builder or regex instead of chaining Replace calls